### PR TITLE
Fix compile errors with MacOS

### PR DESCRIPTION
**Pull request by pwpiwi**
- _POSIX_C_SOURCE must not be defined for num_CPU()

### CODE REVIEW

**Comment by marshmellow42** [comment]
log
```gcc -std=c99 -D_ISOC99_SOURCE -I../../common -I../../client -Wall -O3  -o mfkey32 crypto1.o crapto1.o parity.o util.o mfkey.o mfkey32.c
mfkey32.c: In function 'main':
mfkey32.c:50:3: warning: implicit declaration of function 'msclock' [-Wimplicit-function-declaration]
   uint64_t start_time = msclock();
   ^
C:\Local\Temp\cccOZ8X8.o:mfkey32.c:(.text.startup+0x1b3): undefined reference to `msclock'
C:\Local\Temp\cccOZ8X8.o:mfkey32.c:(.text.startup+0x28f): undefined reference to `msclock'

File: client/util_posix.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// the license.
+//-----------------------------------------------------------------------------
+// utilities requiring Posix library functions
+//-----------------------------------------------------------------------------
+
+#ifndef UTIL_POSIX_H__
+#define UTIL_POSIX_H__
+
+#include "util_posix.h"
+
```
Comment (marshmellow42) [inline client/util_posix.h]: should the header file include itself?

**Comment by marshmellow42** [comment]
appears `#include "util_posix.h"` is needed in mfkey64.c

**Comment by pwpiwi** [comment]
Absolutely correct. And needs a change in tools/Makefile as well.

**Pull request merged**